File: Code Chef/JavaScript/400-499/483 Bucket and Water Flow WATERFLOW.js
process.stdin.setEncoding("utf-8");

const strToInt = (strArray) => {
    strArray.shift();
    const intArray = strArray.map((strItem) => strItem.split(" ").map(Number));
    
    return intArray;
};

const checkBucketStatus = (waterInBucket, bucketCapacity, waterRate, timeInHrs) => {
    while(timeInHrs > 0){
        waterInBucket += waterRate;
        --timeInHrs;
    }
    
    (waterInBucket === bucketCapacity)
        ? console.log("filled")
        : (waterInBucket >= bucketCapacity)
            ? console.log("overflow")
            : console.log("Unfilled");
};

process.stdin.on("data", (input) => {
    const testcases = strToInt(input.trim().split("\n"));
    
    testcases.forEach((testcase) => {
        const [waterInBucket, bucketCapacity, waterRate, timeInHrs] = testcase;
        checkBucketStatus(waterInBucket, bucketCapacity, waterRate, timeInHrs);
    });
});

File: Code Chef/C Programming/400-499/483 Bucket and Water Flow  WATERFLOW.c
#include <stdio.h>
int main() 
{
    int t,i,w,x,y,z;
    scanf("%d",&t);
    for(i=0;i<t;i++)
    {
        scanf("%d%d%d%d",&w,&x,&y,&z);
        if((x-w)==(y*z))
        {
            printf("filled\n");
        }
        else if((x-w)>(y*z))
        {
            printf("Unfilled\n");
        }
        else if((x-w)<(y*z))
        {
            printf("overflow\n");
        }
    }
	return 0;
}